### ISSUE TITLE
no sonor image

### ISSUE THREAD

**Issue by huangbintao188** [Field-Robotics-Lab/dave#278]
I used the git checkout 5344060071d0a72857a9efb558cc1f2d7c0052c6 method and catkin_make succeeded, when i try to run roslaunch nps_uw_multibeam_sonar sonar_tank_blueview_p900_nps_multibeam.launch [https://field-robotics-lab.github.io/dave.doc/contents/dave_sensors/Multibeam-Forward-Looking-Sonar/](url) .My gazebo interface starts and runs normally, but the other window buleview_p900 is dark and has no sonar image. Check the configuration artificialVehicleVibration and it is true. Open rqt to check the sonar image and the values are all 0, which means it is not a computer problem. It may be some configuration error, but I No reason can be found at present. 
![Uploading 14D56A518DD97C1893DB94A1D27FE29A.png…]()

**Comment by gaoning0109** [comment]
Are you solved it ?

**Comment by huangbintao188** [comment]
no, I don't know exactly why. It's the same with docker method.

**Comment by gaoning0109** [comment]
I found the P_Beams in gazebo_multibeam_sonar_ray_based.cpp is empty. Maybe the cuda cal something wrong.

**Comment by woensug-choi** [comment]
Let me get back to you after I finish merging pending PRs first.

**Comment by huangbintao188** [comment]
> I found the P_Beams in gazebo_multibeam_sonar_ray_based.cpp is empty. Maybe the cuda cal something wrong.

What is your CUDA version

**Comment by woensug-choi** [comment]
@huangbintao188 Checked on empty PBeams on CUDA 11.4.. The original code is developed on CUDA 11.1

**Comment by woensug-choi** [comment]
@huangbintao188 Oh! Are you using Docker? or On the Host??
I've tried on Ubuntu 22.04, NVIDIA 545, CUDA 12.3 and it works perfectly fine!
I am guessing mismatch between Docker CUDA version and the host CUDA version.

**Comment by huangbintao188** [comment]
Yes, that should be my version mismatch. If you run well on Ubuntu 22, I can configure it on our server host. Does Dave and np_sonar support ROS2 on Ubuntu 22? How is the specific process for deploying Dave? Thank you

**Comment by huangbintao188** [comment]
@[woensug-choi](https://github.com/woensug-choi)

**Comment by woensug-choi** [comment]
Not for ROS2. I've tested with docker environmont meyhod on Ubuntu 22.04. Sonar worked fine :)

**Comment by Evintjh** [comment]
Hi, I am using NVIDIA 535, docker ubuntu 20.04 and cuda version 11.1. But I am also not seeing any sonar images. Anyone knows why? @huangbintao188 @woensug-choi @gaoning0109

**Comment by woensug-choi** [comment]
for cuda 11.1 the nvidia 535 isnt compatible. Use 450 which is too old...

![image](https://github.com/Field-Robotics-Lab/dave/assets/7955120/d5b1277f-b336-41cb-b410-71a39dc7ab88)

![image](https://github.com/Field-Robotics-Lab/dave/assets/7955120/93782069-7b68-4ce1-b147-984e28c64602)

**Comment by Evintjh** [comment]
Oh, but I tried NVIDIA 535, docker ubuntu 20.04 and cuda version 12.3 too and it gave me similar results @woensug-choi

**Comment by woensug-choi** [comment]
Ive just tried exact same environment yesterday and it worked. Does your host machine and docker has same cuda versions?? you can check by typing nvidia-ami both inside and outside docker

**Comment by huangbintao188** [comment]
I plan to deploy it on a lab computer, it's Ubuntu22, cuda is 12.1, is this environment compatible with Dave's docker method? Do you need me to modify the e.g. docker file? @woensug-choi

**Comment by woensug-choi** [comment]
try docker method and see if cuda version matches inside and outside docker environment

**Comment by woensug-choi** [comment]
Ive just updated the document

**Comment by Evintjh** [comment]
Hi @woensug-choi , when you used Ubuntu 22.04, which version of ROS did you use? I thought Noetic is not compatible with it?

**Comment by Evintjh** [comment]
![Screenshot from 2023-12-07 22-47-24](https://github.com/Field-Robotics-Lab/dave/assets/113495344/c2e9fc15-7350-4e6d-9092-0d69aafa26eb)
![Screenshot from 2023-12-07 22-47-42](https://github.com/Field-Robotics-Lab/dave/assets/113495344/f25e3189-97bb-4eb9-8453-a4ec990fc85f)

Both my host and docker are showing the same driver and cuda version. But I realised that I am using nvidia container toolkit, hence I did not have to physically install cuda toolkit in my docker container. Is there another specific set of instructions if I am using Nvidia container toolkit? @woensug-choi

**Comment by woensug-choi** [comment]
folliwing installation instruction on multibeam sonar document page will do

**Comment by woensug-choi** [comment]
@Evintjh Any updates?

**Comment by aimas-lund** [comment]
@Evintjh upgrading from Nvidia Driver 535, CUDA 12.2 to Driver 545, CUDA 12.3 seemed to solve this issue for me.
Notably my host OS is Ubuntu 22.04.4 LTE, should it be relevant.

**Comment by zxylov** [comment]
@woensug-choi I have followed installation instruction on multibeam sonar document page to configure docker ,but the version of cuda is 12.4,how to change the version to 12.3, or the version of 12.4 can also accept the image ?

**Comment by akgoins** [comment]
I had the same problem as others (black image) even with a newer NVidia/Cuda drivers.  I tried 535, 545, and 550 with no luck.  Turns out the problem was the version of nvcc used to compile the plugin with.  In my docker, the debian of `nvidia-cuda-toolkit` gave nvcc version 11.5 but when I manually installed Cuda toolkitt 12.0 and rebuilt the plugin, the sonar images started working for me again.

**Comment by Aaronl6** [comment]
@akgoins thank you very much. Solved the problem following your advice. Rebuild is important
